metadata-ingestion/src/datahub/ingestion/source/bigquery/bigquery_v2.py

```diff
@@ -0,0 +1,897 @@
+                    _report[SourceCapability.USAGE_STATS] = CapabilityReport(
+                        capable=False,
+                        failure_reason=f"Usage capability test failed with: {e} for project {project_id}",
+                    )
+                    break
+
+            if SourceCapability.USAGE_STATS not in _report:
+                _report[SourceCapability.USAGE_STATS] = CapabilityReport(capable=True)
+
+            test_report.capability_report = _report
+            return test_report
+
+        except Exception as e:
+            test_report.basic_connectivity = CapabilityReport(
+                capable=False, failure_reason=f"{e}"
+            )
+            return test_report
+
+    def is_checkpointing_enabled(self, job_id: JobId) -> bool:
+        if (
+            job_id == self.get_default_ingestion_job_id()
+            and self.is_stateful_ingestion_configured()
+            and self.config.stateful_ingestion
+            and self.config.stateful_ingestion.remove_stale_metadata
+        ):
+            return True
+
+        return False
+
+    def get_default_ingestion_job_id(self) -> JobId:
+        """
+        Keeping the same job_id as for the old source
+        """
+        return JobId("common_ingest_from_sql_source")
+
+    def create_checkpoint(self, job_id: JobId) -> Optional[Checkpoint]:
+        """
+        Create the custom checkpoint with empty state for the job.
+        """
+        assert self.ctx.pipeline_name is not None
+        if job_id == self.get_default_ingestion_job_id():
+            return Checkpoint(
+                job_name=job_id,
+                pipeline_name=self.ctx.pipeline_name,
+                platform_instance_id=self.get_platform_instance_id(),
+                run_id=self.ctx.run_id,
+                config=self.config,
+                state=BaseSQLAlchemyCheckpointState(),
+            )
+        return None
+
+    def get_dataplatform_instance_aspect(
+        self, dataset_urn: str
+    ) -> Optional[MetadataWorkUnit]:
+        # If we are a platform instance based source, emit the instance aspect
+        if self.config.platform_instance:
+            aspect = DataPlatformInstanceClass(
+                platform=make_data_platform_urn(self.platform),
+                instance=make_dataplatform_instance_urn(
+                    self.platform, self.config.platform_instance
+                ),
+            )
+
+            return wrap_aspect_as_workunit(
+                "dataset", dataset_urn, "dataPlatformInstance", aspect, self.report
+            )
+        else:
+            return None
+
+    def get_platform_instance_id(self) -> str:
+        """
+        The source identifier such as the specific source host address required for stateful ingestion.
+        Individual subclasses need to override this method appropriately.
+        """
+        config_dict = self.config.dict()
+        host_port = config_dict.get("host_port", "no_host_port")
```

Comment:
`host_port = self.config.host_port or "no_host_port"`. Don't get the dict form.
Ditto for `database` below.

Reply:
Tbh, host_port will be always empty for BigQuery, database as well will be empty. Is it fine if this will be a simple platform?